velox/connectors/hive/HiveConnectorSplit.cpp

```diff
@@ -0,0 +1,166 @@
+  obj["length"] = length;
+
+  folly::dynamic partitionKeysObj = folly::dynamic::object;
+  for (const auto& [key, value] : partitionKeys) {
+    partitionKeysObj[key] =
+        value.has_value() ? folly::dynamic(value.value()) : nullptr;
+  }
+  obj["partitionKeys"] = partitionKeysObj;
+
+  obj["tableBucketNumber"] = tableBucketNumber.has_value()
+      ? folly::dynamic(tableBucketNumber.value())
+      : nullptr;
+
+  folly::dynamic customSplitInfoObj = folly::dynamic::object;
+  for (const auto& [key, value] : customSplitInfo) {
+    customSplitInfoObj[key] = value;
+  }
+  obj["customSplitInfo"] = customSplitInfoObj;
+  obj["extraFileInfo"] = *extraFileInfo;
+
+  folly::dynamic serdeParametersObj = folly::dynamic::object;
+  for (const auto& [key, value] : serdeParameters) {
+    serdeParametersObj[key] = value;
+  }
+  obj["serdeParameters"] = serdeParametersObj;
+
+  folly::dynamic infoColumnsObj = folly::dynamic::object;
+  for (const auto& [key, value] : infoColumns) {
+    infoColumnsObj[key] = value;
+  }
+  obj["infoColumns"] = infoColumnsObj;
+
+  if (properties.has_value()) {
+    folly::dynamic propertiesObj = folly::dynamic::object;
+    propertiesObj["fileSize"] = properties->fileSize.has_value()
+        ? folly::dynamic(properties->fileSize.value())
+        : nullptr;
+    propertiesObj["modificationTime"] = properties->modificationTime.has_value()
+        ? folly::dynamic(properties->modificationTime.value())
+        : nullptr;
+    obj["properties"] = propertiesObj;
+  } else {
+    obj["properties"] = nullptr;
+  }
+
+  return obj;
+}
+
+// static
+std::shared_ptr<HiveConnectorSplit> HiveConnectorSplit::create(
+    const folly::dynamic& obj) {
+  const auto connectorId = obj["connectorId"].asString();
+  const auto splitWeight = obj["splitWeight"].asInt();
+  const auto filePath = obj["filePath"].asString();
+  const auto fileFormat =
+      dwio::common::toFileFormat(obj["fileFormat"].asString());
+  const auto start = static_cast<uint64_t>(obj["start"].asInt());
+  const auto length = static_cast<uint64_t>(obj["length"].asInt());
+
+  std::unordered_map<std::string, std::optional<std::string>> partitionKeys;
+  for (const auto& [key, value] : obj["partitionKeys"].items()) {
+    partitionKeys[key.asString()] = value.isNull()
+        ? std::nullopt
+        : std::optional<std::string>(value.asString());
+  }
+
+  const auto tableBucketNumber = obj["tableBucketNumber"].isNull()
+      ? std::nullopt
+      : std::optional<int32_t>(obj["tableBucketNumber"].asInt());
+
+  std::unordered_map<std::string, std::string> customSplitInfo;
+  for (const auto& [key, value] : obj["customSplitInfo"].items()) {
+    customSplitInfo[key.asString()] = value.asString();
+  }
+
+  std::shared_ptr<std::string> extraFileInfo =
+      std::make_shared<std::string>(obj["extraFileInfo"].asString());
+  std::unordered_map<std::string, std::string> serdeParameters;
+  for (const auto& [key, value] : obj["serdeParameters"].items()) {
```

Comment:
Do you need to sort based on key? Like https://github.com/facebookincubator/velox/pull/11194 thanks!